app/helpers/application_helper.rb

```diff
@@ -503,6 +503,10 @@ def category_badge(category, opts = nil)
     CategoryBadge.html_for(category, opts).html_safe
   end
 
+  def email_category_badge(category, opts = nil)
+    EmailCategoryBadge.html_for(category, opts).html_safe
+  end
```

Comment:
Could the email styling be added to the existing `CategoryBadge.html_for` helper but only when `inline_style` is enabled? From a quick look at our existing ERB templates, the email templates (digest.html.erb, _mailing_list_post.html.erb` are the only ones using that parameter, so it seems fitting to update the helper for that particular use case. 

Also means long-term we only have one place to worry about.

Reply:
On a closer look, I think the `inline_style` parameter is no longer used in the CategoryBadge helper, even though we are still passing it. Feels like a regression from https://github.com/discourse/discourse/commit/797da5870be1e651f68d32ec47374a057c878625